api/src/test/java/org/pmiops/workbench/db/dao/InstitutionDaoTest.java

```diff
@@ -3,65 +3,192 @@
+    assertThat(savedWithout.getEmailDomains()).isEmpty();
+    assertThat(savedWithout.getEmailAddresses()).isEmpty();
   }
 
   @Test
   public void test_delete() {
-    assertThat(institutionDao.findAll()).hasSize(1);
-    institutionDao.delete(institutionDao.findOneByShortName("Broad").get());
-    assertThat(institutionDao.findAll()).hasSize(0);
+    institutionDao.delete(institutionNoEmailAddressesOrDomains.getInstitutionId());
+    DbInstitution dbInstitution =
+        institutionDao.findOne(institutionNoEmailAddressesOrDomains.getInstitutionId());
+    assertThat(dbInstitution).isNull();
+    assertThat(institutionDao.findAll()).containsExactly(institutionWithEmailAddressesAndDomains);
   }
 
   @Test
   public void test_findAll() {
-    assertThat(institutionDao.findAll()).containsExactlyElementsIn(ImmutableList.of(testInst));
-
-    DbInstitution otherInst = new DbInstitution("VUMC", "Vanderbilt");
-    otherInst = institutionDao.save(otherInst);
     assertThat(institutionDao.findAll())
-        .containsExactlyElementsIn(ImmutableList.of(testInst, otherInst));
+        .containsExactly(
+            institutionNoEmailAddressesOrDomains, institutionWithEmailAddressesAndDomains);
+  }
 
-    institutionDao.delete(institutionDao.findOneByShortName("Broad").get());
-    assertThat(institutionDao.findAll()).containsExactlyElementsIn(ImmutableList.of(otherInst));
+  @Test
+  public void test_findOne() {
+    DbInstitution dbInstitution =
+        institutionDao.findOne(institutionNoEmailAddressesOrDomains.getInstitutionId());
+    assertThat(dbInstitution).isEqualTo(institutionNoEmailAddressesOrDomains);
+    assertThat(dbInstitution.getEmailDomains()).isEmpty();
+    assertThat(dbInstitution.getEmailAddresses()).isEmpty();
+
+    dbInstitution =
+        institutionDao.findOne(institutionWithEmailAddressesAndDomains.getInstitutionId());
+    assertThat(dbInstitution).isEqualTo(institutionWithEmailAddressesAndDomains);
+    assertThat(dbInstitution.getEmailDomains()).isEqualTo(emailDomains);
+    assertThat(dbInstitution.getEmailAddresses()).isEqualTo(emailAddresses);
   }
 
   @Test
   public void test_findOneByShortName() {
-    assertThat(institutionDao.findOneByShortName("Broad")).hasValue(testInst);
+    assertThat(institutionDao.findOneByShortName("Broad"))
+        .hasValue(institutionNoEmailAddressesOrDomains);
+    assertThat(institutionDao.findOneByShortName("Broad1"))
+        .hasValue(institutionWithEmailAddressesAndDomains);
     assertThat(institutionDao.findOneByShortName("Verily")).isEmpty();
+  }
+
+  @Test
+  public void test_updateAllNewEmailAddresses() {
+    Set<DbInstitutionEmailAddress> newEmailAddresses =
+        Sets.newHashSet(
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress1"),
+            new DbInstitutionEmailAddress().setEmailAddress("emailAddress2"));
+    institutionWithEmailAddressesAndDomains.setEmailAddresses(newEmailAddresses);
```

Comment:
Why can't this `setEmailAddresses` simply take in a collection of Strings? It could do the construction and population of the dbInstitution reference in one place so the caller doesn't have to.

Reply:
Now that is an interesting idea.  I'll look into it.

Reply:
This does not appear to be possible with accessor-based Hibernate.

Reply:
☹️ I thought the new hotness was field-based https://pmi-engteam.slack.com/archives/CPC84FQRF/p1580305523002500?thread_ts=1580305523.002500. I guess the jury is still out, but this is one more reason to do it on the properties I would think.

Reply:
I also came to that incorrect conclusion based on that conversation.  I have updated the conversation hopefully to avoid further confusion.